### PR TITLE
[input] Untangle key processing from key translation

### PR DESCRIPTION
**Pull request by garbear**
turn this

``` c++
OnKey(m_Keyboard.ProcessKeyDown(newEvent.key.keysym));
```

into

``` c++
m_Keyboard.ProcessKeyDown(newEvent.key.keysym);
OnKey(m_Keyboard.TranslateKey(newEvent.key.keysym));
```

### CODE REVIEW

File: xbmc/input/KeyboardStat.cpp [hunk 1]
```diff
@@ -177,6 +178,15 @@ const CKey CKeyboardStat::ProcessKeyDown(XBMC_keysym& keysym)
   return key;
 }
 
+void CKeyboardStat::ProcessKeyDown(XBMC_keysym& keysym)
+{
+  if (!(m_lastKeysym == keysym))
+  {
+    m_lastKeysym = keysym;
+    m_lastKeyTime = CTimeUtils::GetFrameTime();
```
Comment (FernetMenta) [inline xbmc/input/KeyboardStat.cpp]: as you are in it you could change GetFrameTime to SystemClockMillis. tying hold time to gui fps makes no sense.

Comment (garbear) [inline xbmc/input/KeyboardStat.cpp]: k, added

**Comment by FernetMenta** [comment]
looks fine to me

**Comment by garbear** [comment]
relatively straightforward. this one's been in my retroplayer tree for over two years now :) jenkins build and merge

**Comment by notspiff** [comment]
you want to cancel this one..

'''
KeyboardStat.cpp: In member function 'CKey CKeyboardStat::TranslateKey(XBMC_keysym&) const':
KeyboardStat.cpp:161:61: warning: statement has no effect [-Wunused-value]
     held = XbmcThreads::SystemClockMillis(); - m_lastKeyTime;
'''

**Comment by garbear** [comment]
jenkins build this please

**Pull request merged**